### PR TITLE
Argspecs for edpm_frr role

### PR DESCRIPTION
**Pull request by jpodivin**

### CODE REVIEW

File: roles/edpm_frr/meta/argument_specs.yml [hunk 1]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: int
+      edpm_frr_bfd_transmit_interval:
+        default: 500
+        description:
+          The minimum transmission interval (less jitter) that this
+          system wants to use to send BFD control packets. (milliseconds)
+        type: int
+      edpm_frr_bgp:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:35]: `description: Enable bgpd daemon`

https://docs.frrouting.org/en/latest/setup.html?highlight=daemons#daemons-configuration-file

File: roles/edpm_frr/meta/argument_specs.yml [hunk 2]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_asn:
+        default: 64999
+        description: |
+          Default BGP ASN. Verify that it is correct for your use case.
+        type: int
+      edpm_frr_bgp_expose_only_host:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:44]: @luis5tb can you provide a description for this?

File: roles/edpm_frr/meta/argument_specs.yml [hunk 3]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_ipv4:
+        default: true
+        description: ''
+        type: bool
+      edpm_frr_bgp_ipv4_allowas_in:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:52]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 4]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_ipv4_src_network:
+        default: ctlplane
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:56]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 5]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_bgp_ipv6:
+        default: true
+        description: ''
+        type: bool
+      edpm_frr_bgp_ipv6_allowas_in:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:64]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 6]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_ipv6_src_network:
+        default: ctlplane
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:68]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 7]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_bgp_l2vpn:
+        default: false
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:72]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 8]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_l2vpn_ebgp_multihop:
+        default: 0
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:76]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 9]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: int
+      edpm_frr_bgp_l2vpn_peers:
+        default:
+        - 192.168.0.1
+        - 192.168.0.2
+        description: List of VPN peer IPv4 addresses.
+        type: list
+      edpm_frr_bgp_l2vpn_peers_scope:
+        default: external
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:86]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 10]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_bgp_l2vpn_uplink_activate:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:90]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 11]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_bgp_learning_routes:
+        default: false
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:94]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 12]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+          The amount is subtracted from packet TTL.
+        type: int
+      edpm_frr_bgp_peers:
+        default: []
+        description: List of IPs or hostnames FRR should peer with
+        type: list
+      edpm_frr_bgp_uplinks:
+        default:
+        - nic1
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:115]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 13]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: list
+      edpm_frr_bgp_uplinks_scope:
+        default: internal
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:119]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 14]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_conf_custom_globals:
+        default: ''
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:123]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 15]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_conf_custom_ipv4:
+        default: ''
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:127]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 16]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_conf_custom_ipv6:
+        default: ''
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:131]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 17]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_conf_custom_router_bgp:
+        default: ''
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:135]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 18]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_config_basedir:
+        default: /var/lib/config-data/ansible-generated/frr
+        description: Path to FRR configuration directory.
+        type: path
+      edpm_frr_defaults:
+        default: traditional
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:143]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 19]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        default: quay.io/podified-antelope-centos9/openstack-frr:current-podified
+        description: URL to openstack-frr container image.
+        type: str
+      edpm_frr_log_level:
+        default: informational
+        description: Log level settings.
+        type: str
+      edpm_frr_log_timestamp_precision:
+        default: 3
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:159]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 20]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: int
+      edpm_frr_version:
+        default: 7.0
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:163]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 21]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        - /etc/pki/tls/cert.pem:/etc/pki/tls/cert.pem:ro
+        - /var/lib/kolla/config_files/frr.json:/var/lib/kolla/config_files/config.json:ro
+        - /var/lib/config-data/ansible-generated/frr:/var/lib/kolla/config_files/src:ro
+        - /var/log/containers/frr:/var/log/frr:z
+        - /run/frr:/run/frr:shared,z
+        description: List of FRR volumes in a mountpoint form.
+        type: list
+      edpm_frr_watchfrr:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:183]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 22]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_watchfrr_options:
+        default: ''
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:187]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 23]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: str
+      edpm_frr_zebra:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:191]: @luis5tb as above :sweat_smile:

File: roles/edpm_frr/meta/argument_specs.yml [hunk 24]
```diff
@@ -3,4 +3,194 @@ argument_specs:
...
+        type: bool
+      edpm_frr_zebra_nht_resolve_via_default:
+        default: true
+        description: ''
```
Comment (rebtoor) [inline roles/edpm_frr/meta/argument_specs.yml:195]: @luis5tb as above :sweat_smile:

**Review by rebtoor — CHANGES_REQUESTED** [review]
@jpodivin i try to gather info for some values, but it's far out of my knowledge here... i mentioned @luis5tb aksing for his help adding description to fields.

**Comment by jpodivin** [comment]
recheck

**Comment by jpodivin** [comment]
> @jpodivin i try to gather info for some values, but it's far out of my knowledge here... i mentioned @luis5tb aksing for his help adding description to fields.

If we don't get any feedback soon I would recommend merging this in it's current state, and filling the details later.

**Comment by jpodivin** [comment]
recheck

**Review by rebtoor — APPROVED** [review]

**Comment by jpodivin** [comment]
recheck

**Comment by jpodivin** [comment]
recheck

**Review by slagle — APPROVED** [review]

**Pull request merged**